### ISSUE TITLE
[proposed enhancement] Workspace Feature  : projects under different colors in sidebar

### ISSUE THREAD

**Issue by RobertoMalatesta** [Microsoft/vscode#31308]
- VSCode Version: Nightly
- OS Version: Any

Workspace feature is really useful as it is today, but could themes API be extended so to display different projects with different colors/nuances? 

See:

![projects_with_colors_in_vsc_workspace](https://user-images.githubusercontent.com/11835025/28514861-dd1a39b0-705b-11e7-9ed3-fca12a8cefe3.png)

It would be even better if tabs could have different colors, to resemble the project they belong to:

![projects_and_tabs_with_colors_in_vsc_workspace](https://user-images.githubusercontent.com/11835025/28514909-062f4fca-705c-11e7-8376-17fbcd1a4130.png)

Thanks for all the beautiful code and the effort you put in this superb project.

Roberto.

**Comment by johnbrett** [comment]
I think this would be an awesome addition - one additional request I would have on this is that we could set colours, or underline to highlight a project's root, as when you have many they are hard to distinguish which is root at a glance, something like:

![image](https://user-images.githubusercontent.com/4015939/31665871-39fdbe26-b342-11e7-9b39-89a5cf4b6f64.png)

If this would be possible, I would be happy to work on a PR for this if someone could give me pointers!

**Comment by ggb667** [comment]
I was surprised this wasn't available already.  I would really like to see this also.

**Comment by jpike88** [comment]
+1 on this. juggling projects in multi-root workspaces needs more distinction between the projects.

**Comment by jpike88** [comment]
This is actually starting drive me insane, I think this should be given way more importance. I also think this color coding should show up in quick search results.

**Comment by malutanpetronel** [comment]
+1 Found this very important ! See Netbeans to find more details on how it works

**Comment by infoflocrm** [comment]
Yes really need this feature. .vscode setting should picked up in this case but its only work when you open as a folder not as workspace.

Tab color should be adjustable via .vscode settings in each project in workspace and one that does not have that settings should use default theme color settings , like same time with workspace folders we open files from ftp. Ftp files should use default settings. Thanks.

**Comment by logicbomb421** [comment]
Yeah this would make a huge difference for me as well. Would love to see this in an upcoming build!

**Comment by rjzheng** [comment]
+9001 this would be provide a much better dev experience!!

**Comment by RuiffCardoso** [comment]
I was looking for this as well, would be great for my multi-workspace setup.

**Comment by nhuesmann** [comment]
I could really use this for an app I'm currently working on, please make it happen VSCode!

**Comment by hdd1013** [comment]
+1

**Comment by schwab** [comment]
Found this while searching the same feature.  So, here's my +1, though I wonder if it could be done through an extension.  Continuing to search...

**Comment by lamnguyen95** [comment]
Hi there,

It'd be really wonderful if VSCode supports this feature! I've switched from Atom and planned to stay with VSCode in next 3 - 5 years ;)

**Comment by invasionofsmallcubes** [comment]
Absolutely agree, I'm having the same issue. I have some configuration files that are the same for different projects, it would be awesome to distinguish them in some way. (you can go with the mouse over the tab to see the path, but it's not really a direct approach).

**Comment by nicolashardy** [comment]
That would be awesome. Does anyone know a workaround in order to get this result?

**Comment by oreporan** [comment]
Hey
I've built a vscode extension that does exactly this (and a bit more)
Built for people who have monorepos or large projects and want to navigate more easily.

its called *suprise* colorTabs

https://marketplace.visualstudio.com/items?itemName=orepor.color-tabs-vscode-ext

**Comment by schwab** [comment]
thank you @oreporan.  Installed it and it's working great.  In my case I set it up thusly: 
`"colorTabs.titleBackground": true,
    "colorTabs.config": [
        {
            "regex": ".*/site-packages/.*",
            "color": "#dd3333",
            "label": "SITE-PACKAGES"
        }
]`

Which turns my title bar Red when I'm editing a  python site-packages files. Since those are overwritten by a build, this is a good way to prevent code loss.  Have already shared with others on my team, they are are :+1:

**Comment by oreporan** [comment]
Glad it works for you - didn't even think about that use-case! makes sense to add some label for `build/`, `dir/` `out/` etc!

**Comment by malutanpetronel** [comment]
...
    "colorTabs.config": [
        { "regex": ".*/site/app/.*", "color": "#dd3333", "label": "app" },
        { "regex": ".*/site/src/.*", "color": "#991111", "label": "src" }
    ]

for some reason this config does not apply anymore..
However somehow the site/aoo have all a red background at top which cannot be changed

**Comment by oreporan** [comment]
@malutanpetronel It would be better to post this under https://github.com/oreporan/color-tabs-vscode/issues/
But in your case, its probably that the regex is wrong, did you mean to do `.*/site/app/.*` ?

**Comment by malutanpetronel** [comment]
yes

**Comment by oreporan** [comment]
@malutanpetronel so did changing the regex work for you? if not you can open an issue and I'll take a look, though its pretty straight forward, if the file/folder matches the regex, it will work

**Comment by malutanpetronel** [comment]
something not ok... now even if I uninstall colortabs extension, the title bar remains red

**Comment by oreporan** [comment]
the extension changes the color by modifying the titleBar
Go to workspace settings and look for `workbench.colorCustomizations`
You can delete that line completely

**Comment by 1138-4EB** [comment]
> I think this would be an awesome addition - one additional request I would have on this is that we could set colours, or underline to highlight a project's root, as when you have many they are hard to distinguish which is root at a glance, something like:

Ref #74229.

> I've built a vscode extension that does exactly this (and a bit more)
> Built for people who have monorepos or large projects and want to distinguish files/folders easily.
>
> It can color the tab/title-bar and add a label based on a regex that you provide

Ref #76399.

**Comment by LucaGabi** [comment]
please also read this feature request here: #95182 in my opinion it is similar request with less clutter

**Comment by jdreina7** [comment]
Hi! There's any news about this feature??? We really need it pls @VSCodeTriageBot

**Comment by MattyMags** [comment]
Why is this still not a thing yet. Its been 4 years. So aggravating.

**Comment by dilirity** [comment]
Hey! Just found this thread. Sad to see it's been 6 years and this hasn't shipped.

There are a few extensions, but they are limited.

Any ETA on this?

**Comment by dgtvan** [comment]
The extension ColorTabs is great, however, what it provides differs from what is described in this thread.

Still looking for a way to color project files using different colours.

**Comment by McDusty** [comment]
I would like to upvote this feature request. 
 ColorTabs looks like a cool extension but it only works on windows. Mac users are are out luck.
 I would very much like to color my tabs to denote important files I want to reference.

**Comment by neilyoung** [comment]
> well i am using this extension working well for me
> 
> https://marketplace.visualstudio.com/items?itemName=orepor.color-tabs-vscode-ext

Seems to not work on Mac

**Comment by neilyoung** [comment]
My two cent here: It is rediculous to see that such a useful feature request remains unresolved for that long, while I'm getting  updates every month, I will never ever use in my life.

**Comment by nullbio** [comment]
Just for record, #213003 is not a duplicate of this issue and is referring to filename colors based on a regexp match (for example, applying text color to all hidden files via `\.+`, not projects.

**Comment by dimateos** [comment]
> Wanted to summarize a bit the status of this long-pending issue. Best regards!

# Coloring tabs: existing extensions

* ColorTabs: https://marketplace.visualstudio.com/items?itemName=orepor.color-tabs-vscode-ext

This solution is very nice for a visual reference when you are inside potentially dangerous sources etc.
But does nothing to inactive tabs nor file tree text area.
I guess it resorts to editing local config files dynamically based on current active file path.

* Tabs color: https://marketplace.visualstudio.com/items?itemName=mondersky.tabscolor

This one can edit the style of inactive tabs
![image](https://github.com/microsoft/vscode/assets/26868611/0e433a86-9c63-49a1-af19-b700fa8e0abe)

Both extensions are a nice start and one could contribute to it, but it would be much preferable if they were built-in.
For example, the second one is severely lacking:

* Due to a lack of vscode API, it resorts to editing internal CSS to change inactive tabs: this triggers a warning about corrupt vscode install the first time.
* Cannot change tabs based on regex: only filetype and ABSOLUTE parent path (not relative nor partial).

# Coloring filenames in file explorer: an example

> I guess there is no API for this, I could not find any extension.

See the following example. IntelliJ has a limited predefined set of groups, nonetheless I would implement a general solution for any file/folder.
![image](https://github.com/microsoft/vscode/assets/26868611/a91250c4-9319-4878-af88-099efabf4fcb)

# Conclusion

Overall it seems like vscode is lacking the required API / settings: 
* **Style any tab based on regex**
  * ATM there is styling only for the active tab, which I guess should not override style based on regex but overlay on it.
  * **Recently [custom labels](https://code.visualstudio.com/docs/getstarted/userinterface#_customize-tab-labels) were introduced in vscode**. This feature already lets the user replace the TEXT of any tab for a custom one based on REGEX, I guess setting the style is the next logical step.
* **Style text in file explorer based on regex**
  * This would require more work... I believe there are no similar settings / features already in vscode.

In both cases, I would expect at least customizable text and bg color, but there could be more style properties like font family, size, etc. Moreover, there could also be many feature around them (e.g. context menu to set tmp color), but I think a static setting should be first.

**Comment by yap-paf** [comment]
+1

**Comment by mattkenefick** [comment]
These were marked as duplicates of this one:

- [Colored labels / layers in the tree view ](https://github.com/microsoft/vscode/issues/200047)
- [Add labels/colors for folders ](https://github.com/microsoft/vscode/issues/62104)
- [Allow custom text/filename color in explorer ](https://github.com/microsoft/vscode/issues/213003)

**Comment by mattmc3** [comment]
This feature ties closely with this discussion as well: (https://github.com/microsoft/vscode-mssql/issues/1684).

With the retirement of Azure Data Studio next February and the push towards using the microsoft/vscode-mssql extension instead, there's a renewed need for a build-in tab colorization feature that would allow extensions to indicate context. In a traditional codebase, it might not be that important to have this feature, but in the context of using VSCode to develop SQL against different databases, this simple feature becomes essential to help distinguish different tabs operating in different contexts.

The scope of this feature as defined (colorizing multiple UI elements) might be too large for the feature needed for microsoft/vscode-mssql, so that might be why this feature has lingered open here for nearly a decade - if it makes more sense to spin off tab colorization to its own issue so that it's a more manageable feature to get traction on, let's do that.

**Comment by iamkarlson** [comment]
There's an extension that allows you to colorize folders and files names based on a path but it doesn't allow to colorize background in the explorer. Here's an issue where it was requested https://github.com/egorovsa/vscode-folder-colorizer/issues/10 It would be awesome to have such an API in VSCode so we can finally have colors like in JetBrains IDEs

**Comment by egorovsa** [comment]
I am still waiting for my feature request. [feature-request ](https://github.com/microsoft/vscode/issues/239308) 

Once it's implemented I will be able to update my [folder-color](https://marketplace.visualstudio.com/items?itemName=SergeyEgorov.folder-color) extension to support background coloring. 

At the moment this is not possible because `vscode.FileDecoration`  does not provide us the way to set a background color through its API

**Comment by collagerai** [comment]
For folks still subscribed and using Claude Code — built a small workaround that paints VS Code window chrome per project (title bar, tab strip, status bar, active-tab top accent). It writes `workbench.colorCustomizations` into `.vscode/settings.json` (and into `.code-workspace` files when present), so each window stands out in Alt+Tab.

Not the sidebar tint OP asked for (the sidebar token is reused by some panels so I deliberately kept it neutral), but the rest of the chrome makes per-project distinction work today.

Repo: https://github.com/collagerai/claude-code-colorize-project — MIT, Win/Mac/Linux.
